### DISCUSSION TITLE
modified tutorial to run MD of bulk water in NVT

### DISCUSSION THREAD

**Discussion by gabriele16** [mir-group/pair_nequip#26] [General]
Hi,

Thanks for the very interesting work and package. I thought I'd share a colab tutorial for a short MD simulation of water that I slightly modified from the nice colab example on toluene (https://bit.ly/mrs-nequip), as I have found some problems when running this colab tutorial. I had similar issues as those shown in https://github.com/mir-group/pair_nequip/issues when trying to compile lammps or when running an md simulation in nvt. I followed some of the guidelines provided in the issues to adapt the tutorial to the bulk water system. The tutorial is for a box of liquid water with 32 molecules resulting from a DFT-MD simulation with CP2K. The DFT forces and positions files in xyz format are converted to extxyz that can be fed to the .yaml configuration file. The colab example runs fine, at least for now with the current version of nequip. I downgraded to a previous pytorch version. In case this is useful, I include the link below, I hope it's ok. Not sure if it fits more in the issues or somewhere else... https://github.com/gabriele16/nequip/blob/main/colab/my-short-nequip-tutorial.ipynb

**Comment by Linux-cpp-lisp** [comment]
Hi @gabriele16 ,

Thank you for your interest in our code, and for sharing your example notebook!

> as I have found some problems when running this colab tutorial

Can you elaborate? We want to ensure that the tutorial is working correctly... did you have issues running it unmodified in Colab, or only once you changed it to use your water dataset? It's not clear to me from your link to the general issues page which problems you were encountering.

> I downgraded to a previous pytorch version.

Was there a specific reason you found this necessary?

Thanks again for sharing this with everyone!

**Reply by gabriele16** [reply]
Hi @Linux-cpp-lisp 

Thanks for the reply! 

Sorry, I should have been more specific and include the posts that discussed some of the issues, I hope it's ok and will be a bit clearer now:

Mainly, I wanted to make an example to run on nvt and encountered this issue: https://github.com/mir-group/pair_nequip/issues/15#issuecomment-1082123872 

However, even when running the unchanged tutorial of toluene, compiling Lammps fails. Does the original tutorial currently run fine on colab? I wonder what I may be doing wrong in that case. I compiled with the runtime set on the gpu (standard) on colab and got this error.
See image below:
<img width="900" alt="Screenshot 2022-07-09 at 23 32 48" src="https://user-images.githubusercontent.com/5607492/178123372-462c9fcc-1b1e-48a6-89e3-e4871a2deaf9.png">

On the original toluene tutorial I tried to downgrade to pytorch 1.10 and changed the compilation of lammps and now it runs for me. I include the tutorial on toluene with these changes here, just in case. https://github.com/gabriele16/nequip/blob/main/mod_nequip_tutorial.ipynb

As suggested in the README.md https://github.com/mir-group/pair_nequip, when cloning lammps I have changed to
`! git clone -b stable_29Sep2021_update2 --depth 1 git@github.com:lammps/lammps`
whereas in the original toluene tutorial the most recent lammps version is downloaded as:
`! wget "https://github.com/lammps/lammps/archive/stable.zip"`

Even after changing to the previous version of Lammps, downgrading to pytorch 1.10 was necessary. If pytorch is not downgraded to version 1.10 lammps compilation fails with the error below. 

<img width="1301" alt="Screenshot 2022-07-10 at 11 43 03" src="https://user-images.githubusercontent.com/5607492/178139537-a41de327-0a40-4f64-901f-7f69bf1dbed5.png">

Not sure if the current pytorch version on colab (1.11) is supported by nequip, according to https://github.com/mir-group/pair_nequip/discussions/17#discussioncomment-2556084

Another minor change in the original colab tutorial, deployment may have to be changed to include the --train-dir option.
`!nequip-deploy build --train-dir results/toluene/example-run-toluene toluene-deployed.pth`

Thanks a lot!
